python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -121,16 +122,28 @@ def broadcast(*columns: Column, target_length: int | None = None) -> list[Column
     ]
 
 
-@dataclasses.dataclass
-class IR:
+class IR(Node):
     """Abstract plan node, representing an unevaluated dataframe."""
 
+    __slots__ = ("schema",)
+    _non_child: ClassVar[tuple[str, ...]] = ("schema",)
```

Comment:
It seemed somewhat broken to me that mypy would require you to specify the type here, but it seems like mypy in fact does not support inheriting this kind of specification at all. [Is this a bug](https://mypy-play.net/?mypy=latest&python=3.12&gist=05adbbda581a906036086f9c7020ae79)? Wouldn't this make A and B in compatible in my example? Do we need to report upstream? I'm further confused because below classes like `PythonScan` do not specify the type, which my example seems to indicate means that nothing would be enforced on `PythonScan`'s definition of `_non_child`.

Reply:
Thanks for following up!